### PR TITLE
update cluster tests to use den launcher

### PR DESCRIPTION
**Pull request by BelSasha**

### CODE REVIEW

File: tests/fixtures/on_demand_cluster_fixtures.py [hunk 1]
```diff
@@ -66,6 +79,25 @@ def ondemand_aws_docker_cluster(request):
...
+@pytest.fixture(scope="session")
+def den_launched_ondemand_aws_docker_cluster(request):
+    """
+    Note: Also used to test docker and default env with alternate Ray version.
+    """
+    args = {
+        "name": "aws-cpu",
+        "instance_type": "CPU:2+",
+        "provider": "aws",
+        "image_id": "docker:rayproject/ray:latest-py311-cpu",
```
Comment (jlewitt1) [inline tests/fixtures/on_demand_cluster_fixtures.py:91]: isn't this failing when running locally? we don't have image / image_id wired up yet to send to Den for launching

Comment (BelSasha) [reply tests/fixtures/on_demand_cluster_fixtures.py:96]: actually works pretty well! :)
attached is an output of `pytest --level release tests -k "den_launched_ondemand_aws_docker_cluster" --detached` (failures are not related to the cluster init)

image

File: tests/test_resources/test_clusters/test_on_demand_cluster.py [hunk 2]
```diff
@@ -87,10 +87,14 @@ class TestOnDemandCluster(tests.test_resources.test_clusters.test_cluster.TestCl
     RELEASE = {
         "cluster": [
             "ondemand_aws_docker_cluster",
+            "den_launched_ondemand_aws_docker_cluster",
             "ondemand_gcp_cluster",
             "ondemand_aws_https_cluster_with_auth",
             "ondemand_k8s_cluster",
             "ondemand_k8s_docker_cluster",
+            # TODO [SB]: un-mark once launching k8 cluster via den is fully supported
+            # "den_launched_ondemand_aws_k8s_cluster",
```
Comment (jlewitt1) [inline tests/test_resources/test_clusters/test_on_demand_cluster.py:96]: launching with eks should work already, we can test this

File: tests/fixtures/on_demand_cluster_fixtures.py [hunk 3]
```diff
@@ -129,6 +165,44 @@ def ondemand_k8s_cluster(request):
...
+    kube_config_path = Path.home() / ".kube" / "config"
+
+    if not kube_config_path.exists():
+        pytest.skip("no kubeconfig found")
+
+    args = {
+        "name": "k8s-cpu-den",
+        "provider": "kubernetes",
+        "instance_type": "CPU:1",
+        "memory": ".2",
```
Comment (jlewitt1) [inline tests/fixtures/on_demand_cluster_fixtures.py:179]: dont think we can specify both memory and instance for "kubernetes"

**Review by jlewitt1 — APPROVED** [review]

**Pull request merged**